presto-main/src/main/java/com/facebook/presto/SystemSessionProperties.java

```diff
@@ -896,7 +897,17 @@ public SystemSessionProperties(
                         ALLOW_WINDOW_ORDER_BY_LITERALS,
                         "Allow ORDER BY literals in window functions",
                         featuresConfig.isAllowWindowOrderByLiterals(),
-                        false));
+                        false),
+                booleanProperty(
+                        EMPTY_JOIN_SOURCES_OPTIMIZATION,
+                        "Simplify joins with one or more empty sources",
+                        featuresConfig.isEmptyJoinOptimization(),
+                        true));
```

Comment:
I think the general style we follow is to have it off by default, turn it on in config and default to true after a couple of releases if there are no issues.